student/academicyear.php

```diff
@@ -51,29 +52,46 @@
 $varsadmin = $siteadmin ? 'yes' : 'no';
 $vararchiveinst = $archivedinstance ? 'yes' : 'no';
 
-echo '<form method="POST" id="yearform" action="">'
+// Get the config data.
+$config = get_config('report_myfeedback');
+
+// Show the academic year menu.
+ $o = '<form method="POST" id="yearform" action="">'
         . '<input type="hidden" name="sesskey" value="' . sesskey() . '" />'
-        . "<input type=\"hidden\" name=\"archive\" value=\"no\"><select id=\"mySelect\" value=$res name=\"myselect\">";
+        . "<input type=\"hidden\" name=\"archive\" value=\"no\">"
+        . "<select id=\"mySelect\" value=$res name=\"myselect\">";
+$i = 0;
 foreach ($acyears as $val) {
-    echo "<option value=" . $val;
-    if ($res == $val) {
-        echo ' selected';
-    }
     if ($val == 'current') {
-        echo ">" . $val;
+        $url = 'current';
+        $linktext = (isset($config->academicyeartext) && $config->academicyeartext != '' ? $config->academicyeartext :
+            get_string('current_academic_year', 'report_myfeedback'));
     } else {
-        echo ">" . "20" . implode('/', str_split($val, 2));
+        $alink = "archivelink" . $i;
+        $url = $config->$alink;
+        $alt = 'archivelinktext' . $i;
+        $linktext = (isset($config->$alt) ? $config->$alt : '');
+    }
+
+    if ($url != '' && $linktext != '') {
+        $o .= "<option url='" . $url . "' value=" . $val .
+            ($val == $res ? ' selected' : '') .
+            ">" . $linktext;
     }
+    $i++;
 }
-echo "</select></form>";
+
+$o .= "</select></form>";
 
 // No support before academic year 1213.
 if ($res != 'current' && $res < 1213) {
-    echo get_string('noarchivesupporth1', 'report_myfeedback');
-    echo get_string('noarchivesupporth2', 'report_myfeedback');
+    $o .= get_string('noarchivesupporth1', 'report_myfeedback');
+    $o .= get_string('noarchivesupporth2', 'report_myfeedback');
     $_SESSION['viewyear'] = 'current';
 }
 
+echo $o;
```

Comment:
why can't you echo directly out of interest?

Reply:
oh, that was one of the 'coding styles' I adopted at QM, probably just that. And it allows to easier follow up the buildup during debugging... do you want me to revert that?

Reply:
would be worth bringing to a Dev Talk. 
Doesn't sound right to me but Alex is on leave so will let it slide for now